freedesktop.org.xml.in

```diff
@@ -6316,4 +6320,15 @@ command to generate the output files.
     <glob pattern="*.stl"/>
   </mime-type>
 
+<mime-type type="application/x-gcode">
+    <_comment>G-code file</_comment>
+    <sub-class-of type="text/plain"/>
+    <generic-icon name="text-x-generic"/>
+    <magic priority="10">
+      <match type="string" value="; " offset="0"/>
+      <match type="string" value="( " offset="0"/>
```

Comment:
The gcode and cnc files I know typically start with a comment stating the name of the file, the date it was created,... I copied this "behavior" from the C source files entry which include "//" and "/*" as a way of identifying them.